src/app/components/sidebar/sidebar.component.html

```diff
@@ -85,12 +84,13 @@
             <a [routerLink]="['/reports/general']">General Reports</a>
           </li>
             <li class="has-sub"
+                id="sidebar-reports-other"
                 routerLinkActive="active"
                 (click)="$event.stopPropagation(); expandedMenu = 'reports/other'"
                 [ngClass]="{'expand': expandedMenu === ('reports/other') }">
             <a href="javascript:"><b class="caret pull-right"></b> Other Reports</a>
             <ul class="sub-menu">
-              <li routerLinkActive="active">
+              <li routerLinkActive="active" id="sidebar-reports-other-all">
```

Comment:
As you mentioned, adding an ID just to facilitate testing is not ideal.
Consider doing something like:
```
by.xpath("//li[@routerLinkActive][contains(text(),'Report All')]");
or maybe
by.xpath("//li[@routerLinkActive && contains(text(),'Report All')]");

or if those don't work:
by.xpath("//li[contains(text(),'Report All')]");
```

I believe it will pick up the text content of any child elements, so you should be good.